src/NextGen/DonationForm/Blocks/DonationFormBlock/resources/app/DonationFormBlockApp.tsx

```diff
@@ -3,27 +3,33 @@ import getDefaultValuesFromSections from './utilities/getDefaultValuesFromSectio
 import Form from './form/Form';
 import {GiveDonationFormStoreProvider} from './store';
 import getWindowData from './utilities/getWindowData';
+import prepareFormData from './utilities/PrepareFormData';
+import getJoiRulesForForm from './utilities/ConvertFieldAPIRulesToJoi';
 
 /**
  * Get data from the server
  */
 const {form} = getWindowData();
 
+prepareFormData(form);
```

Comment:
Memory preservation for such a large object. I didn't see any reason to do this immutably.

Reply:
Oh I see, it's just assigning the `requiredLabel` property to fields.  What else do you anticipate doing in here? Wondering if this function should stay kind of generic or be named more specific since at first glance it's not clear what exactly it's  _preparing_

Reply:
This now also adds node handling functions to the form and its groups in 9ad0deb.